Assets/Scripts/FactionController.cs

```diff
@@ -0,0 +1,244 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A class to hold references to all the factions 
+/// provides methods to access and modify the factions
+/// </summary>
+public class FactionController : MonoBehaviour
+{
+
+    /// <summary>
+    /// The list of factions
+    /// </summary>
+    /// <typeparam name="Faction">Individual faction objects</typeparam>
+    /// <returns></returns>
+     [SerializeField] List<Faction> factions = new List<Faction>();
+
+    /// <summary>
+    /// Reference to the game master
+    /// </summary>
+     GameMaster gameMaster;
+
+    void Start()
+    {
+        //get the game master
+        gameMaster = GetComponent<GameMaster>();
+        
+    }
+
+    /// <summary>
+    /// Gets the power of all factions
+    /// </summary>
+    /// <returns>a float array of the faction powers</returns>
+    public float[] GetFactionPower()
+    {
+        float[] factionPower = new float[factions.Count];
+        for (int i = 0; i < factions.Count; i++)
+        {
+            factionPower[i] = factions[i].FactionPower;
+        }
+        return factionPower;
+    }
+
+    /// <summary>
+    /// Gets the happiness of all factions
+    /// </summary>
+    /// <returns>
+    /// a float array of the faction happiness
+    ///</returns>
+    public float[] GetFactionHappiness()
+    {
+        float[] factionHappiness = new float[factions.Count];
+        for (int i = 0; i < factions.Count; i++)
+        {
+            factionHappiness[i] = factions[i].FactionHappiness;
+        }
+        return factionHappiness;
+    }
+
+    /// <summary>
+    /// Gets the happiness of all factions with the AI
+    /// </summary>
+    /// <returns>
+    /// a float array of the faction happiness with the AI
+    /// </returns>
+    public float[] GetFactionAiHappiness()
+    {
+        float[] factionAiHappiness = new float[factions.Count];
+        for (int i = 0; i < factions.Count; i++)
+        {
+            factionAiHappiness[i] = factions[i].FactionAiHappiness;
+        }
+        return factionAiHappiness;
+    }
+
+    /// <summary>
+    /// Given a faction name, returns the faction object
+    /// </summary>
+    /// <param name="factionName">
+    /// The name of the faction to get
+    /// </param>
+    /// <returns>
+    ///    The faction object or null if the faction name is not found
+    /// </returns>
+    public Faction SelectFaction(string factionName)
+    {
+        for (int i = 0; i < factions.Count; i++)
+        {
+            if (factions[i].FactionName == factionName)
+            {
+                return factions[i];
+            }
+        }
+        return null;
```

Comment:
Add a warning/breakpoint here if we fail to find one.
Probably a breakpoint cause it's serious if we fail.

Reply:
Unless we're removing factions when they hit 0.

Reply:
log and breakpoint added.

Reply:
Nice